### PR TITLE
StatusBar imperative API: honor animation defaults for optionals

### PR DESCRIPTION
**Pull request by mosesoak**
StatusBar's imperative API contains optional animation inputs in the 
methods `setHidden`, `setBarStyle` and `setBackgroundColor`, but not 
passing values currently defaults to no animation, even when a 
StatusBar instance at the top of the app declares animation props. 

Since these animation inputs are declared as optional and no method 
for setting up global defaults is provided, from a developer 
perspective, it makes sense to honor the current props stack’s 
animation settings.

For example if this is included at the top of the app,

`<StatusBar animated={true} showHideTransition='slide' />`

it intuitively makes most sense that calling `StatusBar.setHidden(true)` 
would still show a slide animation, instead of needing to pass `'slide'`.

A slightly more architecture-correct way to fix this would be to add a 
`StatusBar.setDefaultProps(props:Object)` method, but that would 
require changing the API and docs. The current docs seem to hint that 
it is possible to set defaults on a top-level StatusBar instance, so it 
feels simplest to just honor those animation defaults when defined.

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
By analyzing the blame information on this pull request, we identified @janicduplessis and @qpowell to be potential reviewers.

File: Libraries/Components/StatusBar/StatusBar.js [hunk 1]
```diff
@@ -136,8 +136,11 @@ const StatusBar = React.createClass({
     // Provide an imperative API as static functions of the component.
     // See the corresponding prop for more detail.
     setHidden(hidden: boolean, animation?: StatusBarAnimation) {
-      animation = animation || 'none';
-      StatusBar._defaultProps.hidden.value = hidden;
+      StatusBar._defaultProps.hidden = hidden;
+      if (!animation) {
```
Comment (mosesoak) [inline Libraries/Components/StatusBar/StatusBar.js]: Note: this input is a string so a non-strict inequality check is used. The inputs below are booleans so strict inequality is used so `false` will be honored.

**Comment by mosesoak** [comment]
Food for thought: the current docs do indicate that the imperative API should not be mixed with the declarative API, so maybe this PR is not the right approach. Perhaps simpler would be to document the imperative API methods and indicate that it is passing animation parameters is required to see animation when using these methods, all declarative settings will be ignored.

Or both: we could allow mixing, document the imperative API and indicate that declarative animation settings will be honored. I did not change the docs but would be willing to do that for this PR if that's the direction decided on. Thanks!

**Comment by janicduplessis** [comment]
@mosesoak Thanks for the PR!

I think I'd rather keep the imperative API as separated as possible from the component. Having a function that it's default value changes over time may be confusing. Maybe we should just document the default values better?

**Comment by mosesoak** [comment]
Okay sounds good. I think what would make it clearest and easiest to use would be:

1. List out the imperative API methods in the docs (currently they're only shown in examples so things like how defaults are handled is not clear)
2. Since I don't like having to pass a string like `'fade'` every time I call `setHidden`, it would be nice to add a `StatusBar.setDefaultProps(props:Object)` method that could be called once during app setup.

Happy to help with either or both of those if you want, thanks for the quick response on this PR!

**Comment by janicduplessis** [comment]
Component methods are now displayed on the website (it only appears in version 0.25+ of the docs) http://facebook.github.io/react-native/releases/0.25/docs/statusbar.html#content

At the moment there is no documentation for the component methods so we could add a docblock that says pretty much the same thing as the prop and add what the default value is.

I'm good with adding `setDefaultProps` to change the `_defaultProps` value.

**Comment by mosesoak** [comment]
Okay cool. I'll make a new PR when I get time, unless you want to do any of this

**Pull request closed without merging**